SparkPost/2web2ui
# FE-1192: Add the Change Report Modal to dashboard

FE-1192: Add the Change Report Modal to dashboard

### What Changed

- Modified the existing ReportsListModal
      - moved in into src/components/reportBuilder
      - used the Tabs from src/components/matchbox as the other Tabs component was defaulting to the My Reports Tab on any re-render 
      - Added a `onDashboard` prop to this Modal to add support for https://sparkpost.invisionapp.com/console/Dashboard_PinnedReport-ckfydojkz00cr01vsbf9z1pt4/ckfydokg600cu01vs3qyndqlx/play

     - on Dashboard, on opening this Modal, the Submit button should be disabled when no checkbox is selected
     - on Dashboard, on submit of this modal, `pinned_report` user ui option should be set with correct report id.


### How To Test
- Use the Appteam account on staging to get to this onboarding step.
- On Clicking the “Change Report” button, this [modal](https://sparkpost.invisionapp.com/console/Dashboard_PinnedReport-ckfydojkz00cr01vsbf9z1pt4/ckfydokg600cu01vs3qyndqlx/play) Dashboard_PinnedReport  should open up
- On clicking “Change Report” save the selection as a User UI Option. - “pinned_report”
- Check if the ReportsList Modal works as earlier on the ReportBuilder Page.


### To Do

- [x] Fix this issue - Modal is resetting when you search for a report name and select the checkbox.
